### DISCUSSION TITLE
Boot ipxe.efi to shell without using embedded script and Ctrl-B

### DISCUSSION THREAD

**Discussion by ccy** [ipxe/ipxe#733] [General]
Hi,

My machine has multiple network interfaces.

When I execute `ipxe.efi` from UEFI built-in shell manually, the `ipxe.efi` will perform DHCP for all network interface automatically.

The Ctrl+B prompt flashing too fast, I have tried several times and never get into the prompt successfully.  It direct start DHCP command after I press Ctrl-B.

Is that a way to `ipxe.efi` and direct go into shell without using embedded script?

**Comment by NiKiZe** [comment]
You can open the shell with the shell command, so you need:

#!ipxe
shell

**Reply by ccy** [reply]
So this is embedding a script?  I would like to use the `ipxe.efi` in multiple cases and trying to avoid making `ipxe.efi` for each use case.

**Comment by Feng-zero** [comment]
I also encountered the same problem. In uefi mode, when entering ipxe boot on the bootoveride interface, although it displayed the need to input ctrl-b, I was unable to enter the ipxe shell and instead started auto boot. Is there any restriction on UEFI mode?

**Reply by mcb30** [reply]
Check that you are using an up-to-date iPXE.  There is a UEFI design defect that prevents the Ctrl key from working properly: we worked around this in commit https://github.com/ipxe/ipxe/commit/a08244ecc

**Reply by Feng-zero** [reply]
Thank you for your reply. I will update the code and try it out
